test/testdata/lsp/hover_union.rb

```diff
@@ -0,0 +1,29 @@
+# typed: true
+extend T::Sig
+
+class A
+  extend T::Sig
+
+  # Docs for A#foo
+  sig {returns(String)}
+  def foo; ''; end
+end
+
+class B
+  extend T::Sig
+
+  # Docs for B#foo
+  sig {returns(Integer)}
+  def foo; 0; end
+end
+
+sig {params(x: T.any(A, B)).void}
+def example(x)
+  x.foo
+  # ^ hover: ```ruby
+  # ^ hover: sig {returns(T.any(String, Integer))}
+  # ^ hover: def foo; end
+  # ^ hover: ```
+  # ^ hover: ---
+  # ^ hover: Docs for A#foo
```

Comment:
Currently this doesn't work as expected since hover assertions operate on a per line basis, with no support for multi-line hover. Is there a recommended approach for achieving multiline?

Reply:
This test passes locally for me, and editing one of the lines here triggers test failures.  `test/testdata/lsp/hover.rb` contains a similar multi-line matching testcase.